docs/advanced/experimental/async-executors.md

```diff
@@ -0,0 +1,141 @@
+(docker-compose)=
+# Using async python within Executors
+
+
+You can use naturally call async coroutines within `Executor`'s, allowing you to leverage the power of asynchronous
+python to write concurrent code. 
+
+
+````{admonition} Example code
+:class: tip
+
+Functions decorated by `requests`  can be directly implemented as async `coroutines. 
+
+```python
+from jina import Executor, requests, Flow
+
+class MyExecutor(Executor):
+    
+    @requests
+    async def encode(self,docs,*kwargs):
+        await some_coroutines()
+```
+````
+
+
+
+## Examples
+
+In this example we have a heavy lifting API for which we want to call several time, we want to leverage the
+async python features to speed up the `Executor`'s call by calling the api multiples times in parallel.
+
+### With async
+
+let's make this api call a python `coroutines`
+
+```python
+import asyncio
+
+async def api_call(text):
+    await asyncio.sleep(1)
+    return text.upper()
+```
+
+and let's modify our code to leverage the non blocking api call
+
+
+```python
+import time
+import asyncio
+from docarray import Document, DocumentArray
+from jina import Executor, requests, Flow
+
+def on_done(resp):
+
+    for doc in resp.docs:
+        assert doc.text == doc.text.upper()
+
+    results = resp.parameters['__results__']
+
+    for res in results.values():
+        print(f"the call took {res['time']} second")
+
+class DummyAsyncExecutor(Executor):
+
+    async def proces_doc(self, doc: Document):
+        doc.text = await api_call(doc.text)
+
+    @requests
+    async def encode_async(self, docs: DocumentArray, **kwargs):
+
+        start_time = time.time()
+        task = [asyncio.ensure_future(self.proces_doc(doc)) for doc in docs]
```

Comment:
Do not need to create tasks, just call a simple `await`. We do not want to show all the `async` features of `python`